### ISSUE TITLE
Duplicated strings are allowed while original is still being sent

### ISSUE THREAD

**Issue by SafaAlfulaij** [Tatoeba/tatoeba2#1626]
Sometimes while translating my internet lags, and the translation submitting loader keeps spinning, so I just go and press the translation icon and then "Add translation" again, assuming it will be faster (sorry server).
It seems that the server can't check duplication here and just allow it.
Examples (the Arabic translations):
https://tatoeba.org/ara/sentences/show/6983814
https://tatoeba.org/ara/sentences/show/5869290

**Comment by jiru** [comment]
I’m unable to reproduce the problem locally, so it’s hard to solve.

It’s probable due to a race condition with two sessions checking duplicate at the very same time. If it’s true, then one way to solve that problem is to use a locking mechanism. It looks like InnoDB has some [cool locking facilities we could use](https://dba.stackexchange.com/questions/116193/lock-mysql-table-to-prevent-select-and-insert-at-the-same-time).

**Comment by agrodet** [comment]
Even if this problem happens, Horus will take care of the sentences later, right? I don't know how the duplicate check is done when adding a sentence, but since it's an irregular situation that doesn't lead to a unexpected situation, probably that's not a big deal. I faced the "forever spinning" problem many times in the past, but I can't remember if I have faced it recently... Except during exceptional period like the recent attack, of course.
